src/api/libcellml/units.h

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2015 University of Auckland
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.Some license of other
+*/
+
+#ifndef LIBCELLML_LIBCELLML_UNITS_H
+#define LIBCELLML_LIBCELLML_UNITS_H
+
+#include <string>
+
+#include "libcellml/libcellml_export.h"
+#include "libcellml/types.h"
+#include "libcellml/namedentity.h"
+
+namespace libcellml {
+
+/**
+ * @brief The Units class.
+ * Class for Units.
+ */
+class LIBCELLML_EXPORT Units: public NamedEntity
+{
+public:
+    Units(); /**< Constructor */
+    ~Units(); /**< Destructor */
+    Units(const Units &rhs); /**< Copy constructor */
+    Units(Units &&rhs); /**< Move constructor */
+    Units& operator=(Units n); /**< Assignment operator */
+
+    /**
+     * @brief Test to determine if Units is a base unit.
+     * Test to determine if Units is a base unit, return True if it is
+     * a base unit and False otherwise.
+     * @return True if Units is a base unit, False otherwise.
+     */
+    bool isBaseUnit() const;
+
+    /**
+     * @brief Set whether this unit is a base unit or not.
+     * Set the base unit flag to signify whether this Units is a base unit
+     * or not.  This flag cannot be set if this Units is a compound unit.
+     * @param state boolean value to set the base unit flag to.
+     */
+    void setBaseUnit(bool state=true);
+
+    /**
+     * @brief Add a unit to this Units.
+     * Add a unit as a child of this Units.  This method takes optional arguments
+     * scale, and offset.
+     * @param units The name of the unit to add.
+     * @param prefix The prefix for the unit, one of CELLML_PREFIXES.
+     * @param exponent The exponent.
+     * @param scale The scale.
+     * @param offset The offset.
+     */
+    void addUnit(const std::string &units, CELLML_PREFIXES prefix, double exponent,
+                 double scale=1.0, double offset=0.0);
+
+    /**
+     * @brief Add a unit to this Units.
```

Comment:
I was thinking that this would allow users to add units that were not standard units like fahrenheit etc. so it is not just for CellML standard units.